ui/src/settings/segmenters/components/form/validation/schema.js

```diff
@@ -0,0 +1,99 @@
+/* eslint-disable no-template-curly-in-string */
+import * as yup from "yup";
+
+import { typeOptions } from "settings/segmenters/components/typeOptions";
+
+const nameRegex = /^[A-Za-z\d][\w\d \-()#$%&:.]*[\w\d\-()#$%&:.]$/;
+const nameRegexDescription =
+  "Name must begin with an alphanumeric character and have no trailing spaces and can contain letters, numbers, blank spaces and the following symbols: -_()#$%&:.";
+const typeOptionsValues = typeOptions.map((e) => e.value);
+
+// Note: The validation functions below are specifically not defined using the
+// arrow format, for access to `this` from the invocation context:
```

Comment:
This comment shouldn't be here. Seems like copy-paste error :) We are not defining any special functions in this file.